lsp-vhdl.el

```diff
@@ -21,44 +21,84 @@
 
 ;;; Commentary:
 
-;; LSP support for VHDL using the language server from http://www.vhdltool.com/download.
-;; Set the following symbol to specify the path of the language server:
-;;
-;; (setq lsp-vhdl-server-path "/path/to/server_binary")
+;; LSP support for VHDL using using an external language server. Currently
+;; the supported servers are:
 ;;
+;; VHDL-tool from http://www.vhdltool.com/download
 ;; The server requires a project file list called vhdltool-config.yaml which must
 ;; be placed the the project root. An example file can be found here:
 ;; http://www.vhdltool.com/configuration
+;;
+;; HDL Checker from https://github.com/suoto/hdl_checker.
+;; This server requires a config file described here
+;; https://github.com/suoto/hdl_checker/wiki/setting-up-a-project
+;;
+;; Set the symbol lsp-vhdl-server to select the language server and set
+;; lsp-vhdl-server-path if the binary is not in the user PATH.
 
 ;;; Code:
 
 (require 'lsp-mode)
 
+(defvar vhdl-tool-bin-name "vhdl-tool"
+  "Name of the VHDL Tool binary.")
+
+(defvar vhdl-tool-disp-name "lsp-vhdl-tool"
+  "Display name for VHDL-tool.")
+
+(defvar hdl-checker-bin-name "hdl_checker"
+  "Name of HDL Checker binary.")
+
+(defvar hdl-checker-disp-name "lsp-hdl-checker"
+  "Display name for HDL Checker.")
+
 (defgroup lsp-vhdl nil
-  "LSP support for VHDL using the server from http://www.vhdltool.com"
-  :group 'lsp-mode
-  :link '(url-link "http://www.vhdltool.com"))
+  "LSP support for VHDL. Set lsp-vhdl-server to select server. The default is to use VHDL-tool."
+  :group 'lsp-mode)
+
+(defcustom lsp-vhdl-server 'vhdl-tool
+  "Select which server to use:
+VHDL-tool: A syntax checking, type checking and linting tool (http://vhdltool.com).
+HDL Checker: A wrapper for third party tools such as GHDL, ModelSim, Vivado Simulator (https://github.com/suoto/hdl_checker)."
+  :type '(choice (const :tag "VHDL-tool" vhdl-tool)
+                 (const :tag "HDL Checker" hdl-checker))
+  :group 'lsp-vhdl)
 
-(defcustom lsp-vhdl-server-path
-  "NOT_SET"
-  "Path to binary server file downloaded from http://www.vhdltool.com/download"
+(defcustom lsp-vhdl-server-path nil
+  "Path to binary server file."
   :group 'lsp-vhdl
   :risky t
   :type 'file)
 
 (defun lsp-vhdl--create-connection ()
-  "Returns lsp-stdio-connection or an error if server not found"
-  (lsp-stdio-connection
-   (lambda ()
-     (list lsp-vhdl-server-path "lsp"))))
+  "Returns lsp-stdio-connection based on the selected server"
+  (setq params nil)
```

Comment:
Having `params` defined globally is not ok for elisp(you may have collisions). You may:

1. Change [ara,s to local - use let and change it to hashtable. 
2. Create a defvar which is prefixed with `lsp-vhdl--` to avoid the collisions. 
3. Do not have the params and calculate the command on demand(e. g. return functions :test? and :connect which will calculate the command on-demand two times.

Reply:
Right, I'll change it to a defvar with name `lsp-vhdl--params` instead - when I figure out how to update a PR with a new commit.